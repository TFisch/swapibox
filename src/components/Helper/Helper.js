import {
  fetchHomeWorld,
  fetchPopulation,
  fetchSpecies,
  fetchVehicleData
} from '../../data/FetchApi';

export const getOpeningScrawl = data => {
  const openingScrawl = {
    title: data.title,
    episode: data.episode_id,
    scrawl: data.opening_crawl
  };
  return openingScrawl;
};

export const cleanHomeworld = peopleData => {
  let wholePeople;
  const unresolvedPeopleData = peopleData.results.map(async person => {
    const name = person.name;

    const response = await person.homeworld;
    const homeWorld = await fetchHomeWorld(response);
    const population = await fetchPopulation(response);
    const species = await fetchSpecies(person.species);
    return (wholePeople = {
      ...person,
      name,
      Homeworld: homeWorld,
      Population: population,
      Species: species
    });
  });
  return Promise.all(unresolvedPeopleData);
};

export const cleanSpecies = peopleData => {
  const unresolvedSpeciesData = peopleData.map(person => {
    const { name, Homeworld, Population } = person;

    return fetch(person.species)
      .then(response => response.json())
      .then(species => ({
        name,
        Homeworld,
        Population,
        Species: species.name,
        isFavoriteSelected: false
      }));
  });
  return Promise.all(unresolvedSpeciesData);
};

export const cleanVehicles = vehicleData => {
  const unresolvedVehicleData = vehicleData.results.map(vehicle => {
    let newVehicle;
    const { name, model, vehicle_class, passengers } = vehicle;
    return (newVehicle = {
      name, model, vehicle_class, passengers
    })

  });
  console.log(unresolvedVehicleData);
  return Promise.all(unresolvedVehicleData);
};
