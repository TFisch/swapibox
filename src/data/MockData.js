export const MockData = {
    people: {
            "count": 87,
            "next": "https://swapi.co/api/people/?page=2",
            "previous": null,
            "results": [{
                        "name": "Luke Skywalker",
                        "height": "172",
                        "mass": "77",
                        "hair_color": "blond",
                        "skin_color": "fair",
                        "eye_color": "blue",
                        "birth_year": "19BBY",
                        "gender": "male",
                        "homeworld": "https://swapi.co/api/planets/1/",
                        "films": [
                            "https://swapi.co/api/films/2/",
                            "https://swapi.co/api/films/6/",
                            "https://swapi.co/api/films/3/",
                            "https://swapi.co/api/films/1/",
                            "https://swapi.co/api/films/7/"
                        ],
                        "species": [
                            "https://swapi.co/api/species/1/"
                        ],
                        "vehicles": [
                            "https://swapi.co/api/vehicles/14/",
                            "https://swapi.co/api/vehicles/30/"
                        ],
                        "starships": [
                            "https://swapi.co/api/starships/12/",
                            "https://swapi.co/api/starships/22/"
                        ],
                        "created": "2014-12-09T13:50:51.644000Z",
                        "edited": "2014-12-20T21:17:56.891000Z",
                        "url": "https://swapi.co/api/people/1/"
                    },
                    {
                        "name": "C-3PO",
                        "height": "167",
                        "mass": "75",
                        "hair_color": "n/a",
                        "skin_color": "gold",
                        "eye_color": "yellow",
                        "birth_year": "112BBY",
                        "gender": "n/a",
                        "homeworld": "https://swapi.co/api/planets/1/",
                        "films": [
                            "https://swapi.co/api/films/2/",
                            "https://swapi.co/api/films/5/",
                            "https://swapi.co/api/films/4/",
                            "https://swapi.co/api/films/6/",
                            "https://swapi.co/api/films/3/",
                            "https://swapi.co/api/films/1/"
                        ],
                        "species": [
                            "https://swapi.co/api/species/2/"
                        ],
                        "vehicles": [],
                        "starships": [],
                        "created": "2014-12-10T15:10:51.357000Z",
                        "edited": "2014-12-20T21:17:50.309000Z",
                        "url": "https://swapi.co/api/people/2/"
                    },
                    {
                        "name": "R2-D2",
                        "height": "96",
                        "mass": "32",
                        "hair_color": "n/a",
                        "skin_color": "white, blue",
                        "eye_color": "red",
                        "birth_year": "33BBY",
                        "gender": "n/a",
                        "homeworld": "https://swapi.co/api/planets/8/",
                        "films": [
                            "https://swapi.co/api/films/2/",
                            "https://swapi.co/api/films/5/",
                            "https://swapi.co/api/films/4/",
                            "https://swapi.co/api/films/6/",
                            "https://swapi.co/api/films/3/",
                            "https://swapi.co/api/films/1/",
                            "https://swapi.co/api/films/7/"
                        ],
                        "species": [
                            "https://swapi.co/api/species/2/"
                        ],
                        "vehicles": [],
                        "starships": [],
                        "created": "2014-12-10T15:11:50.376000Z",
                        "edited": "2014-12-20T21:17:50.311000Z",
                        "url": "https://swapi.co/api/people/3/"
                    },
                    {
                        "name": "Darth Vader",
                        "height": "202",
                        "mass": "136",
                        "hair_color": "none",
                        "skin_color": "white",
                        "eye_color": "yellow",
                        "birth_year": "41.9BBY",
                        "gender": "male",
                        "homeworld": "https://swapi.co/api/planets/1/",
                        "films": [
                            "https://swapi.co/api/films/2/",
                            "https://swapi.co/api/films/6/",
                            "https://swapi.co/api/films/3/",
                            "https://swapi.co/api/films/1/"
                        ],
                        "species": [
                            "https://swapi.co/api/species/1/"
                        ],
                        "vehicles": [],
                        "starships": [
                            "https://swapi.co/api/starships/13/"
                        ],
                        "created": "2014-12-10T15:18:20.704000Z",
                        "edited": "2014-12-20T21:17:50.313000Z",
                        "url": "https://swapi.co/api/people/4/"
                    }
                ]
                
}}

export const films = { title: 'A New Hope',
    episode_id: 4,
    opening_crawl: 'It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire\'s\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire\'s\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....' }
    
export const vehicles = { 
	count: 39,
	next: 'https://swapi.co/api/vehicles/?page=2',
	previous: null,
	results:
	 [ { name: 'Sand Crawler',
			 model: 'Digger Crawler',
			 manufacturer: 'Corellia Mining Corporation',
			 cost_in_credits: '150000',
			 length: '36.8',
			 max_atmosphering_speed: '30',
			 crew: '46',
			 passengers: '30',
			 cargo_capacity: '50000',
			 consumables: '2 months',
			 vehicle_class: 'wheeled',
			 pilots: [],
			 films: [Array],
			 created: '2014-12-10T15:36:25.724000Z',
			 edited: '2014-12-22T18:21:15.523587Z',
			 url: 'https://swapi.co/api/vehicles/4/' },
		 { name: 'T-16 skyhopper',
			 model: 'T-16 skyhopper',
			 manufacturer: 'Incom Corporation',
			 cost_in_credits: '14500',
			 length: '10.4',
			 max_atmosphering_speed: '1200',
			 crew: '1',
			 passengers: '1',
			 cargo_capacity: '50',
			 consumables: '0',
			 vehicle_class: 'repulsorcraft',
			 pilots: [],
			 films: [Array],
			 created: '2014-12-10T16:01:52.434000Z',
			 edited: '2014-12-22T18:21:15.552614Z',
			 url: 'https://swapi.co/api/vehicles/6/' },
		 { name: 'X-34 landspeeder',
			 model: 'X-34 landspeeder',
			 manufacturer: 'SoroSuub Corporation',
			 cost_in_credits: '10550',
			 length: '3.4',
			 max_atmosphering_speed: '250',
			 crew: '1',
			 passengers: '1',
			 cargo_capacity: '5',
			 consumables: 'unknown',
			 vehicle_class: 'repulsorcraft',
			 pilots: [],
			 films: [Array],
			 created: '2014-12-10T16:13:52.586000Z',
			 edited: '2014-12-22T18:21:15.583700Z',
			 url: 'https://swapi.co/api/vehicles/7/' },
		 { name: 'TIE/LN starfighter',
			 model: 'Twin Ion Engine/Ln Starfighter',
			 manufacturer: 'Sienar Fleet Systems',
			 cost_in_credits: 'unknown',
			 length: '6.4',
			 max_atmosphering_speed: '1200',
			 crew: '1',
			 passengers: '0',
			 cargo_capacity: '65',
			 consumables: '2 days',
			 vehicle_class: 'starfighter',
			 pilots: [],
			 films: [Array],
			 created: '2014-12-10T16:33:52.860000Z',
			 edited: '2014-12-22T18:21:15.606149Z',
			 url: 'https://swapi.co/api/vehicles/8/' },
		 { name: 'Snowspeeder',
			 model: 't-47 airspeeder',
			 manufacturer: 'Incom corporation',
			 cost_in_credits: 'unknown',
			 length: '4.5',
			 max_atmosphering_speed: '650',
			 crew: '2',
			 passengers: '0',
			 cargo_capacity: '10',
			 consumables: 'none',
			 vehicle_class: 'airspeeder',
			 pilots: [Array],
			 films: [Array],
			 created: '2014-12-15T12:22:12Z',
			 edited: '2014-12-22T18:21:15.623033Z',
			 url: 'https://swapi.co/api/vehicles/14/' },
		 { name: 'TIE bomber',
			 model: 'TIE/sa bomber',
			 manufacturer: 'Sienar Fleet Systems',
			 cost_in_credits: 'unknown',
			 length: '7.8',
			 max_atmosphering_speed: '850',
			 crew: '1',
			 passengers: '0',
			 cargo_capacity: 'none',
			 consumables: '2 days',
			 vehicle_class: 'space/planetary bomber',
			 pilots: [],
			 films: [Array],
			 created: '2014-12-15T12:33:15.838000Z',
			 edited: '2014-12-22T18:21:15.667730Z',
			 url: 'https://swapi.co/api/vehicles/16/' },
		 { name: 'AT-AT',
			 model: 'All Terrain Armored Transport',
			 manufacturer: 'Kuat Drive Yards, Imperial Department of Military Research',
			 cost_in_credits: 'unknown',
			 length: '20',
			 max_atmosphering_speed: '60',
			 crew: '5',
			 passengers: '40',
			 cargo_capacity: '1000',
			 consumables: 'unknown',
			 vehicle_class: 'assault walker',
			 pilots: [],
			 films: [Array],
			 created: '2014-12-15T12:38:25.937000Z',
			 edited: '2014-12-22T18:21:15.714673Z',
			 url: 'https://swapi.co/api/vehicles/18/' },
		 { name: 'AT-ST',
			 model: 'All Terrain Scout Transport',
			 manufacturer: 'Kuat Drive Yards, Imperial Department of Military Research',
			 cost_in_credits: 'unknown',
			 length: '2',
			 max_atmosphering_speed: '90',
			 crew: '2',
			 passengers: '0',
			 cargo_capacity: '200',
			 consumables: 'none',
			 vehicle_class: 'walker',
			 pilots: [Array],
			 films: [Array],
			 created: '2014-12-15T12:46:42.384000Z',
			 edited: '2014-12-22T18:21:15.761584Z',
			 url: 'https://swapi.co/api/vehicles/19/' },
		 { name: 'Storm IV Twin-Pod cloud car',
			 model: 'Storm IV Twin-Pod',
			 manufacturer: 'Bespin Motors',
			 cost_in_credits: '75000',
			 length: '7',
			 max_atmosphering_speed: '1500',
			 crew: '2',
			 passengers: '0',
			 cargo_capacity: '10',
			 consumables: '1 day',
			 vehicle_class: 'repulsorcraft',
			 pilots: [],
			 films: [Array],
			 created: '2014-12-15T12:58:50.530000Z',
			 edited: '2014-12-22T18:21:15.783232Z',
			 url: 'https://swapi.co/api/vehicles/20/' },
		 { name: 'Sail barge',
			 model: 'Modified Luxury Sail Barge',
			 manufacturer: 'Ubrikkian Industries Custom Vehicle Division',
			 cost_in_credits: '285000',
			 length: '30',
			 max_atmosphering_speed: '100',
			 crew: '26',
			 passengers: '500',
			 cargo_capacity: '2000000',
			 consumables: 'Live food tanks',
			 vehicle_class: 'sail barge',
			 pilots: [],
			 films: [Array],
			 created: '2014-12-18T10:44:14.217000Z',
			 edited: '2014-12-22T18:21:15.807906Z',
			 url: 'https://swapi.co/api/vehicles/24/' } ] }