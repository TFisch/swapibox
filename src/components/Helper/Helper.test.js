import React from 'react'
import {cleanHomeworld, cleanPlanetData} from './Helper'
import { shallow } from 'enzyme'
import {films} from '../../data/MockData.js'

describe('cleanHomeWorld', () => {

  it.only('should take in the correct data', () => {
    let expected = {
      title: films.title,
      episode: films.episode,
      scrawl: films.scrawl
    }
    getOpeningScrawl(films)
    expect(getOpeningScrawl).toEqual(expected)
  })
})