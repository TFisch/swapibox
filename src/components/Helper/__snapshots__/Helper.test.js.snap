// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Helper Method getOpeningScrawl should return correct Data 1`] = `undefined`;

exports[`Helper should match the snapshot 1`] = `undefined`;
